client/lib/analytics/statsd-utils.ts

```diff
@@ -1,20 +1,34 @@
 import config from '@automattic/calypso-config';
 
-function generateUrl( featureSlug: string, eventType: string, eventValue: string ) {
-	const slug = featureSlug.replace( /[.:-]/g, '_' );
-	const type = eventType.replace( '-', '_' );
-	const json = JSON.stringify( {
-		beacons: [ `calypso.${ config( 'boom_analytics_key' ) }.${ slug }.${ type }:${ eventValue }` ],
-	} );
-
-	const [ encodedSlug, jsonData ] = [ slug, json ].map( encodeURIComponent );
-	return `https://pixel.wp.com/boom.gif?v=calypso&u=${ encodedSlug }&json=${ jsonData }`;
+interface BeaconData {
+	name: string;
+	value: number;
+	type: 'counting' | 'timing';
 }
 
-export function statsdTimingUrl( featureSlug: string, eventType: string, duration: number ) {
-	return generateUrl( featureSlug, eventType, `${ duration }|ms` );
+function createBeacon( calypsoSection: string, { name, value, type }: BeaconData ) {
+	const event = name.replace( '-', '_' );
+
+	// A counting event defaults to incrementing by one.
+	if ( type === 'counting' && value == null ) {
+		value = 1;
```

Comment:
Are we using `createBeacon` anywhere where the `value` would be really `null`? I'm asking because the `BeaconData` type doesn't allow that, it says `value` is a `number`.

Reply:
Good point -- I'll look again at the usages and update the types if needed. I think we could have a union type which allows null only for `counting` events

Reply:
I was curious if
```js
if ( type === 'counting' ) {
  value ??= 1;
}
```
also works and it does! I never used this new operator before 🙂

Reply:
Glad to see we're using the [logical nullish assignment operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment) but I hope we don't get to mutate variables too often with it 😅

Reply:
Missed this conversation! Wouldn't it only be useful for mutating variables? E.g. you have an existing variable, and need to set it to something only if hasn't been set yet?